sdk/servicebus/service-bus/src/serviceBusError.ts

```diff
@@ -0,0 +1,179 @@
+import { isMessagingError, MessagingErrorCodes, MessagingError, translate } from "@azure/core-amqp";
+import { AmqpError } from "rhea-promise";
+
+/**
+ * Service Bus failure reasons.
+ */
+export type ServiceBusErrorReason =
+  // note: This list is intended to mirror https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/servicebus/Azure.Messaging.ServiceBus/src/Primitives/ServiceBusFailureReason.cs
+  /**
+   * The exception was the result of a general error within the client library.
+   */
+  | "GeneralError"
+  /**
+   * A Service Bus resource cannot be found by the Service Bus service.
+   */
+  | "MessagingEntityNotFound"
+  /**
+   * The lock on the message is lost. Callers should call attempt to receive and process the message again.
+   */
+  | "MessageLockLost"
+  /**
+   * The requested message was not found.
+   */
+  | "MessageNotFound"
+  /**
+   * A message is larger than the maximum size allowed for its transport.
+   */
+  | "MessageSizeExceeded"
+  /**
+   * An entity with the same name exists under the same namespace.
+   */
+  | "MessagingEntityAlreadyExists"
+  /**
+   * The Messaging Entity is disabled. Enable the entity again using Portal.
```

Comment:
Is this the same as the entity-status? If so, users can use the AdminClient to enable.

Reply:
That's an interesting idea - do we collectively refer to our client and the ARM client as 'management clients'?